tasks/main.yml

```diff
@@ -17,26 +12,328 @@
       {{ ansible_facts['distribution'] ~ '_' ~
       ansible_facts['distribution_version'] }}.yml
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __mssql_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __mssql_vars_file is file
+
+- name: "Verify that the user accepts EULA variables"
+  assert:
+    that:
+      - mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula
+      - mssql_accept_microsoft_cli_utilities_for_sql_server_eula
+      - mssql_accept_microsoft_sql_server_2019_standard_eula
+    fail_msg:
+      - "You must accept EULA by setting the following variables to true:"
+      - "mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula"
+      - "mssql_accept_microsoft_cli_utilities_for_sql_server_eula"
+      - "mssql_accept_microsoft_sql_server_2019_standard_eula"
+  register: __mssql_eula_evaluation
+
+- name: Configure the Microsoft SQL Server 2019 repository
+  yum_repository:
+    name: packages-microsoft-com-mssql-server-2019-gdr
+    description: Microsoft SQL Server 2019
+    baseurl: "{{ __mssql_server_repository }}"
+    gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+    gpgcheck: true
 
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: Ensure that the Microsoft SQL Server package is installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __mssql_server_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-    mode: 0400
-  notify: template handler to restart services
+- name: Check if the errorlogfile setting has a custom value
+  command: /opt/mssql/bin/mssql-conf get filelocation errorlogfile
+  register: __mssql_errorlogfile
+  ignore_errors: true
+  changed_when: false
+
+- name: Check if the errorlog file exists in a custom location
+  stat:
+    path: "{{ __mssql_errorlogfile.stdout | regex_search('[^: ]*$') }}"
+  register: __mssql_errorlog_custom
+  when: "'errorlogfile' in __mssql_errorlogfile.stdout"
+
+- name: Check if the errorlog file exists in the default location
+  stat:
+    path: /var/opt/mssql/log/errorlog
+  register: __mssql_errorlog_default
+  when: "'errorlogfile' not in __mssql_errorlogfile.stdout"
+
+- name: Gather system services facts
+  service_facts:
+
+- name: Verify that the variables required for setting up MSSQL are defined
+  assert:
+    that:
+      - mssql_password is defined
+      - mssql_edition is defined
+    fail_msg:
+      - "You must define the following variables to set up MSSQL Server:"
+      - "mssql_password"
+      - "mssql_edition"
+  when: >
```

Comment:
If the argument to `when` is a `list` then the default operation is `and` - so you can write this as
```yaml
when:
  - '"running" not in ansible_facts["services"]["mssql-server.service"]["state"]'
  - '"enabled" not in ansible_facts["services"]["mssql-server.service"]["status"]'
  - ((__mssql_errorlog_custom.stat is defined) and
    (not __mssql_errorlog_custom.stat.exists)) or
    ((__mssql_errorlog_default.stat is defined) and
    (not __mssql_errorlog_default.stat.exists))
```

Reply:
I suggest you use a `set_fact` to define a variable like `__mssql_not_setup`:
```yaml
- name: Determine if MSSQL was previously set up
  set_fact:
    __mssql_not_setup: "{{
      ('running' not in
      ansible_facts['services']['mssql-server.service']['state']) and
      ('enabled' not in
      ansible_facts['services']['mssql-server.service']['status']) and
      (((__mssql_errorlog_custom.stat is defined) and
       (not __mssql_errorlog_custom.stat.exists)) or
      ((__mssql_errorlog_default.stat is defined) and
       (not __mssql_errorlog_default.stat.exists)))
    }}"

- name: Verify that the variables required for setting up MSSQL are defined
  assert:
    that:
      - mssql_password is defined
      - mssql_edition is defined
    fail_msg:
      - "You must define the following variables to set up MSSQL Server:"
      - "mssql_password"
      - "mssql_edition"
  when: __mssql_not_setup | bool
```